lib/src/component_declaration/component_base/component_base_2.dart

```diff
@@ -72,25 +72,55 @@ abstract class UiComponent2<TProps extends UiProps> extends react.Component2
   /// > Should be used alongside [forwardingClassNameBuilder].
   ///
   /// > Related [copyUnconsumedDomProps]
+  ///
+  /// __Deprecated.__ Use [addUnconsumedProps] within
+  /// [modifyProps] (rather than [addProps]) instead. Will be removed in the
+  /// `4.0.0` release.
   @override
+  @Deprecated('4.0.0')
   Map copyUnconsumedProps() {
     var consumedPropKeys = consumedProps?.map((ConsumedProps consumedProps) => consumedProps.keys) ?? const [];
 
     return copyProps(keySetsToOmit: consumedPropKeys);
   }
 
+  /// Passes a reference of a component's [UiProps] to be updated with unconsumed props.
```

Comment:
We should mention that this should be called as an argument to `modifyProps` - and add an example of doing so:

```suggestion
  /// A prop modifier that passes a reference of a component's `props` to be updated with any unconsumed props.
  ///
  /// Call within `modifyProps` like so:
  ///
  ///     class SomeCompositeComponent extends UiComponent<SomeCompositeComponentProps> {
  ///       @override
  ///       render() {
  ///         return (SomeOtherWidget()..modifyProps(addUnconsumedProps(props)))(
  ///           props.children,
  ///         );
  ///       }
  ///     }
```


I'm also going to update the AC for the codemod (CPLAT-5212) to include:

```diff
- ..addProps(copyUnconsumedProps())
+ ..modifyProps(addUnconsumedProps(props))
```